regression/regression_test_fb.cpp

```diff
@@ -0,0 +1,80 @@
+#include <gtest/gtest.h>
+#include <opendaq/opendaq.h>
+
+using namespace daq;
+
+class RegressionTestFunctionBlock : public testing::TestWithParam<StringPtr>
+{
+private:
+    ModuleManagerPtr moduleManager;
+    ContextPtr context;
+    InstancePtr instance;
+    DevicePtr device;
+
+protected:
+    FunctionBlockPtr fb;
+
+    void SetUp() override
+    {
+        moduleManager = ModuleManager("");
+        context = Context(nullptr, Logger(), TypeManager(), moduleManager);
+
+        instance = InstanceCustom(context, "mock_instance");
+
+        device = instance.addDevice(GetParam());
+
+        // TODO: should not rely on "ref_fb_module_trigger" being present?
```

Comment:
Depends on your test setup - you should probably pre-add some function blocks to be able to test them properly without adding them remotely first.